packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,229 @@
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static slotClassNames: SplitButtonSlotClassNames = {
+    toggleButton: `${SplitButton.className}__toggleButton`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    defaultOpen: PropTypes.bool,
+    menu: PropTypes.oneOfType([
+      customPropTypes.itemShorthand,
+      PropTypes.arrayOf(customPropTypes.itemShorthandWithoutJSX),
+    ]),
+    onMainButtonClick: PropTypes.func,
+    onMenuItemClick: PropTypes.func,
+    onOpenChange: PropTypes.func,
+    open: PropTypes.bool,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+    toggleButton: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+    toggleButton: {},
+  }
+
+  static autoControlledProps = ['open']
+
+  getInitialAutoControlledState(): SplitButtonState {
+    return {
+      isFromKeyboard: false,
+      open: false,
+    }
+  }
+
+  handleMenuButtonOverrides = (predefinedProps: MenuButtonProps) => ({
+    onMenuItemClick: (e: React.SyntheticEvent, menuItemProps: MenuItemProps) => {
+      this.setState({ open: false })
+      _.invoke(this.props, 'onOpenChange', e, { open: false, ...menuItemProps })
+
+      _.invoke(predefinedProps, 'onMenuItemClick', e, menuItemProps)
+      _.invoke(this.props, 'onMenuItemClick', e, menuItemProps)
+    },
+    onOpenChange: (e: React.SyntheticEvent, popupProps: PopupProps) => {
+      e.stopPropagation()
+      this.setState({ open: popupProps.open })
+      _.invoke(this.props, 'onOpenChange', e, popupProps)
+    },
+  })
+
+  handleMenuButtonTriggerOverrides = (predefinedProps: ButtonProps) => ({
+    onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+      _.invoke(predefinedProps, 'onClick', e, buttonProps)
+      _.invoke(this.props, 'onMainButtonClick', e, buttonProps)
+    },
+    onFocus: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+      _.invoke(predefinedProps, 'onFocus', e, buttonProps)
+      this.setState({ isFromKeyboard: isFromKeyboard() })
+    },
+  })
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    styles,
+    unhandledProps,
+  }: RenderResultConfig<MenuButtonProps>): React.ReactNode {
+    const { button, disabled, menu, primary, secondary, toggleButton } = this.props
+    const trigger = Button.create(button, {
+      defaultProps: {
+        styles: styles.button,
+        primary,
+        secondary,
+        disabled,
+      },
+      overrideProps: this.handleMenuButtonTriggerOverrides,
+    })
+
+    return (
+      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
```

Comment:
does this mean people can use onClick and it will be invoked for both click on the button as well as click on the toggle?